### ISSUE TITLE
OnPush changeDetection not working in custom FieldTypes

### ISSUE THREAD

**Issue by skrtheboss** [ngx-formly/ngx-formly#2191]
**Description**
If `ChangeDetectionStrategy.OnPush` is used in a custom FieldType, then the `templateOptions` which are generated via expressionProperties, do not result in a changeDetection check.

I think the problem is [here](https://github.com/ngx-formly/ngx-formly/blob/dad2ef6923188b19f46261cea0db74580e7f95e9/src/core/src/lib/services/formly.form.builder.ts#L68), since ref.changeDetectorRef.markForCheck() does not work as expected (see angular  [issue](https://github.com/angular/angular/issues/18817)).

A possible solution as suggested in this [comment](https://github.com/angular/angular/issues/18817#issuecomment-525833679), would be:
```typescript
if (field._componentRefs) {
    field._componentRefs.forEach(ref => {
          const changeDetectorRef = ref.injector.get(ChangeDetectorRef);
          changeDetectorRef.markForCheck()
    });
}
```

**Minimal Reproduction**
The view should dispaly the updated options, but it does not. If we remove the OnPush strategy, everything works as expected

https://stackblitz.com/edit/angular-vmswdr

**Your Environment**

- Angular version: 9.1.2
- Formly version: .5.5.15

**PS**
If you agree with me I can make a pull request.

Here is a [StackBlitz](https://stackblitz.com/edit/angular-vmswdr-ycccrc) with the fixed changeDetection.

**Comment by aitboudad** [comment]
I'm ok with that change :)

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.6.0](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.6.0) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
